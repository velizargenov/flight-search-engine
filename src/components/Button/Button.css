input[type=button] {
  cursor: pointer;
  background: rgb(168, 168, 168);
  border: 1px solid #000000;
  border-radius: 7px;
  height: 32px;
  width: 124px;
  font-size: 12px;
  letter-spacing: 0.3px;
}

.button--search-this-flight {
  /* border: 2px solid red; */
}