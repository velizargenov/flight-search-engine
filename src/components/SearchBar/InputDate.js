import React from 'react';
import PropTypes from 'prop-types';
import DatePicker from 'react-datepicker';
import moment from 'moment';

const InputDate = ({ text, selectedDate, handleDateChange }) => (
  <div className="search-bar--date">
    <span className="input--text">
      {text}
    </span>
    <span className="search-bar--arrow" />
    <DatePicker
      placeholderText={selectedDate || 'Select a date'}
      minDate={moment('2018-01-17')}
      maxDate={moment('2018-01-17').add(4, 'days')}
      onChange={handleDateChange}
    />
  </div>
);

InputDate.defaultProps = {
  text: 'Return Date',
  selectedDate: '18-01-2018',
  handleDateChange: () => {},
};

InputDate.propTypes = {
  text: PropTypes.string,
  selectedDate: PropTypes.string,
  handleDateChange: PropTypes.func,
};

export default InputDate;
