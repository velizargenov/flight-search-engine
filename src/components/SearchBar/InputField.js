import React from 'react';

const InputField = ({ htmlFor, text, ...otherProps }) => (
  <label htmlFor={htmlFor} className="input">
    <span className="input--text">{text}</span>
    <input id={htmlFor} {...otherProps} />
  </label>
);

export default InputField;
