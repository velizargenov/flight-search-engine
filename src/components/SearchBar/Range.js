import React from 'react';
import PropTypes from 'prop-types';
import InputRange from 'react-input-range';

const Range = ({ rangeValue, handleRangeValueChange }) => (
  <div className="search-bar--input-range">
    <span className="input input--text">Refine Flight Search</span>
    <InputRange
      step={10}
      minValue={0}
      maxValue={200}
      value={rangeValue}
      onChange={handleRangeValueChange}
    />
  </div>
);

Range.defaultProps = {
  handleRangeValueChange: () => {},
  rangeValue: {
    max: 200,
    min: 0,
  },
};

Range.propTypes = {
  handleRangeValueChange: PropTypes.PropTypes.func,
  rangeValue: PropTypes.shape({
    max: PropTypes.number.isRequired,
    min: PropTypes.number.isRequired,
  }),
};

export default Range;
