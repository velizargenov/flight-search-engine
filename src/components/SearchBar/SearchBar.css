.search-bar {
  border-right: 1px solid #000000;
  width: 236px;
  height: 542px;
  padding: 21px 10px;
  float: left;
}
.search-bar form {
  border: 1px solid black;
  min-height: 345px;
}
.input {
  display: inline-block;
  width: 212px; 
  padding: 13px 10px 0px 10px;
  /* width: 216px; */
}
.input input {
  border: 1px solid black;
  width: 208px;
  height: 20px;
  padding: 0 3px
}
.input--text {
  display: block;
  font-size: 12px;
  letter-spacing: 0.5px;
  margin: 0 0 2px 3px;
}
.search-bar--date {
  margin-left: 10px;
  margin-top: 9px;
  position: relative;
}
.search-bar--arrow {
  width: 27px;
  height: 22px;
  position: absolute;
  border-left: 1px solid #000000;
  border-right: 1px solid #000000;
  right: 10px;
  z-index: 999;
  top: 17px;
  background: rgb(207, 207, 207);
  cursor: pointer;
}
.search-bar--arrow:after {
  content: '';
  width: 0;
  height: 0;
  border-left: 8px solid transparent;
  border-right: 8px solid transparent;
  border-top: 10px solid #000000;
  position: absolute;
  top: 6px;
  left: 5px;
}
.search-bar--button-search {
  margin-left: 55px;
  margin: 35px 0 0 55px;
}