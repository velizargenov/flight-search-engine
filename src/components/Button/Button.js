import React from 'react';
import PropTypes from 'prop-types';

import './Button.css';

const Button = ({ htmlFor, handleClick, ...otherProps }) => (
  <label htmlFor={htmlFor} className="button">
    <input
      type="button"
      id={htmlFor}
      onClick={handleClick}
      {...otherProps}
    />
  </label>
);

Button.defaultProps = {
  htmlFor: 'uniqueId',
  handleClick: () => {},
};

Button.propTypes = {
  htmlFor: PropTypes.string,
  handleClick: PropTypes.func,
};

export default Button;
