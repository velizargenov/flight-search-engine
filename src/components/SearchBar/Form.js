import React, { Component } from 'react';
import PropTypes from 'prop-types';

import 'react-datepicker/dist/react-datepicker.css';
import 'react-input-range/lib/css/index.css';
import Button from '../Button/Button';
import InputField from './InputField';
import InputDate from './InputDate';

class Form extends Component {
  constructor (props) {
    super(props);

    this.handlePassengerNumberChange = this.handlePassengerNumberChange.bind(this);
    this.handleFromChange = this.handleFromChange.bind(this);
    this.handleDestinationChange = this.handleDestinationChange.bind(this);
  }
  handlePassengerNumberChange (e) {
    this.props.onPassengerNumberChange(e.target.value);
  }
  handleFromChange (e) {
    this.props.onFromChange(e.target.value);
  }
  handleDestinationChange (e) {
    this.props.onDestinationChange(e.target.value);
  }

  render () {
    const {
      isReturnFlight,
      departureDate,
      returnDate,
      handleDepartureDateChange,
      handleReturnDateChange,
      handleClicksOnSearch,
    } = this.props;

    return (
      <form onSubmit={e => e.preventDefault()}>
        <InputField
          htmlFor="fromInput"
          text="From"
          placeholder="Enter your input here..."
          type="text"
          required
          onChange={this.handleFromChange}
        />
        <InputField
          htmlFor="destinationInput"
          text="Destination"
          placeholder="Enter your input here..."
          type="text"
          required
          onChange={this.handleDestinationChange}
        />
        <InputDate
          selectedDate={departureDate}
          text="Departure Date"
          handleDateChange={handleDepartureDateChange}
        />
        {
          isReturnFlight
            ? <InputDate
              selectedDate={returnDate}
              text="Return Date"
              handleDateChange={handleReturnDateChange}
            />
            : null
        }
        <InputField
          htmlFor="passenger"
          text="Passenger"
          type="number"
          defaultValue="1"
          min="1"
          max="5"
          onChange={this.handlePassengerNumberChange}
        />

        <Button
          htmlFor="search"
          className="search-bar--button-search"
          defaultValue="Search"
          type="submit"
          handleClick={handleClicksOnSearch}
        />

      </form>
    );
  }
}

Form.defaultProps = {
  onPassengerNumberChange: () => {},
  onFromChange: () => {},
  onDestinationChange: () => {},
  isReturnFlight: true,
  departureDate: '18-01-2018',
  returnDate: '19-01-2018',
  handleDepartureDateChange: () => {},
  handleReturnDateChange: () => {},
  handleClicksOnSearch: () => {},
};

Form.propTypes = {
  onPassengerNumberChange: PropTypes.func,
  onFromChange: PropTypes.func,
  onDestinationChange: PropTypes.func,
  isReturnFlight: PropTypes.bool,
  departureDate: PropTypes.string,
  returnDate: PropTypes.string,
  handleDepartureDateChange: PropTypes.func,
  handleReturnDateChange: PropTypes.func,
  handleClicksOnSearch: PropTypes.func,
};

export default Form;
