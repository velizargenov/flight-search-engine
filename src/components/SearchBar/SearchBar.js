import React from 'react';
import PropTypes from 'prop-types';

import './SearchBar.css';
import Button from '../Button/Button';
import Form from './Form';
import Range from './Range';

const SearchBar = (props) => {
  const {
    handleClicksOnOneWayButton,
    handleClicksOnReturnButton,
    handleDepartureDateChange,
    handleReturnDateChange,
    isReturnFlight,
    departureDate,
    returnDate,
    onPassengerNumberChange,
    onFromChange,
    onDestinationChange,
    handleRangeValueChange,
    handleClicksOnSearch,
    rangeValue,
  } = props;

  return (
    <div className="search-bar">
      <div className="search-bar--buttons">
        <Button
          htmlFor="one-way"
          className="button--search-bar one-way"
          value="One way"
          handleClick={handleClicksOnOneWayButton}
        />
        <Button
          htmlFor="return"
          className="button--search-bar return active"
          value="Return"
          handleClick={handleClicksOnReturnButton}
        />
      </div>
      <Form
        handleDepartureDateChange={handleDepartureDateChange}
        handleReturnDateChange={handleReturnDateChange}
        isReturnFlight={isReturnFlight}
        departureDate={departureDate}
        returnDate={returnDate}
        onPassengerNumberChange={onPassengerNumberChange}
        onFromChange={onFromChange}
        onDestinationChange={onDestinationChange}
        handleClicksOnSearch={handleClicksOnSearch}
      />
      <Range
        rangeValue={rangeValue}
        handleRangeValueChange={handleRangeValueChange}
      />
    </div>
  );
};

SearchBar.defaultProps = {
  handleClicksOnOneWayButton: () => {},
  handleClicksOnReturnButton: () => {},
  handleDepartureDateChange: () => {},
  handleReturnDateChange: () => {},
  isReturnFlight: true,
  departureDate: '18-01-2018',
  returnDate: '20-01-2018',
  onPassengerNumberChange: () => {},
  onFromChange: () => {},
  onDestinationChange: () => {},
  handleRangeValueChange: () => {},
  handleClicksOnSearch: () => {},
  rangeValue: {
    max: 200,
    min: 0,
  },
};

SearchBar.propTypes = {
  handleClicksOnOneWayButton: PropTypes.func,
  handleClicksOnReturnButton: PropTypes.func,
  handleDepartureDateChange: PropTypes.func,
  handleReturnDateChange: PropTypes.func,
  isReturnFlight: PropTypes.bool,
  departureDate: PropTypes.string,
  returnDate: PropTypes.string,
  onPassengerNumberChange: PropTypes.PropTypes.func,
  onFromChange: PropTypes.PropTypes.func,
  onDestinationChange: PropTypes.PropTypes.func,
  handleRangeValueChange: PropTypes.PropTypes.func,
  handleClicksOnSearch: PropTypes.PropTypes.func,
  rangeValue: PropTypes.shape({
    max: PropTypes.number,
    min: PropTypes.number,
  }),
};

export default SearchBar;
