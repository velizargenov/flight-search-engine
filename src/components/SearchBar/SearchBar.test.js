import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

import SearchBar from './SearchBar';
import Button from '../Button/Button';
import Form from './Form';
import Range from './Range';
import InputField from './InputField';
import InputDate from './InputDate';

describe('SearchBar Component', () => {
  const wrapper = shallow(<SearchBar />);
  it('matches snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render without crashing', () => {
    expect(wrapper.length).toBe(1);
  });
  it('should render Button, Form, Range components', () => {
    expect(wrapper.containsAllMatchingElements([
      <Button />,
      <Form />,
      <Range />,
    ]));
  });
  it('should render two Button components', () => {
    expect(wrapper.find(Button).length).toBe(2);
  });
  it('should render one of each Form, Range components', () => {
    expect(wrapper.find(Form).length).toBe(1);
    expect(wrapper.find(Range).length).toBe(1);
  });
  it('should have two <div> elements', () => {
    expect(wrapper.find('div').length).toBe(2);
  });
  it('should have one <div> element wrapping the two Button components', () => {
    expect(wrapper.find('div.search-bar--buttons').length).toBe(1);
  });
  it('should have one wrapping <div> element', () => {
    expect(wrapper.find('div.search-bar').length).toBe(1);
  });
});

describe('Form Component', () => {
  const wrapper = shallow(<Form />);
  it('matches snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render without crashing', () => {
    expect(wrapper.length).toBe(1);
  });
  it('should render InputField, InputDate, Button components', () => {
    expect(wrapper.containsAllMatchingElements([
      <InputField />,
      <InputDate />,
      <Button />,
    ]));
  });
  it('should render three InputField components', () => {
    expect(wrapper.find(InputField).length).toBe(3);
  });
  it('should render two InputField components', () => {
    expect(wrapper.find(InputDate).length).toBe(2);
  });
  it('should render one Button component', () => {
    expect(wrapper.find(Button).length).toBe(1);
  });
  it('should have one <form> element', () => {
    expect(wrapper.find('form').length).toBe(1);
  });
});

describe('InputField Component', () => {
  const wrapper = shallow(<InputField />);
  it('matches snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render without crashing', () => {
    expect(wrapper.length).toBe(1);
  });
  it('should have one of each <label>, <span>, <input> elements', () => {
    expect(wrapper.find('label').length).toBe(1);
    expect(wrapper.find('span').length).toBe(1);
    expect(wrapper.find('input').length).toBe(1);
  });
  it('renders text in span element correctly', () => {
    expect(wrapper.find('span').text()).toMatch(/From|Destination/);
  });
});

describe('InputDate Component', () => {
  const wrapper = shallow(<InputDate />);
  it('matches snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render without crashing', () => {
    expect(wrapper.length).toBe(1);
  });
  it('should have one <div> element', () => {
    expect(wrapper.find('div').length).toBe(1);
  });
  it('should have two <span> elements', () => {
    expect(wrapper.find('span').length).toBe(2);
  });
  it('renders text in span element correctly', () => {
    expect(wrapper.find('span.input--text').text()).toMatch(/Departure Date|Return Date/);
  });
  it('should have one wrapping <div> element', () => {
    expect(wrapper.find('div.search-bar--date').length).toBe(1);
  });
});

describe('Range Component', () => {
  const wrapper = shallow(<Range />);
  it('matches snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render without crashing', () => {
    expect(wrapper.length).toBe(1);
  });
  it('should have one <div> element', () => {
    expect(wrapper.find('div').length).toBe(1);
  });
  it('should have one <span> element', () => {
    expect(wrapper.find('span').length).toBe(1);
  });
  it('renders text in span element correctly', () => {
    expect(wrapper.find('span.input--text').text()).toMatch('Refine Flight Search');
  });
  it('should have one wrapping <div> element', () => {
    expect(wrapper.find('div.search-bar--input-range').length).toBe(1);
  });
});
