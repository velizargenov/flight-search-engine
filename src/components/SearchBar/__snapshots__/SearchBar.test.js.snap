// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Component matches snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <InputField
    htmlFor="fromInput"
    onChange={[Function]}
    placeholder="Enter your input here..."
    required={true}
    text="From"
    type="text"
  />
  <InputField
    htmlFor="destinationInput"
    onChange={[Function]}
    placeholder="Enter your input here..."
    required={true}
    text="Destination"
    type="text"
  />
  <InputDate
    handleDateChange={[Function]}
    selectedDate="18-01-2018"
    text="Departure Date"
  />
  <InputDate
    handleDateChange={[Function]}
    selectedDate="19-01-2018"
    text="Return Date"
  />
  <InputField
    defaultValue="1"
    htmlFor="passenger"
    max="5"
    min="1"
    onChange={[Function]}
    text="Passenger"
    type="number"
  />
  <Button
    className="search-bar--button-search"
    defaultValue="Search"
    handleClick={[Function]}
    htmlFor="search"
    type="submit"
  />
</form>
`;

exports[`InputDate Component matches snapshot 1`] = `
<div
  className="search-bar--date"
>
  <span
    className="input--text"
  >
    Return Date
  </span>
  <span
    className="search-bar--arrow"
  />
  <DatePicker
    allowSameDay={false}
    dateFormat="L"
    dateFormatCalendar="MMMM YYYY"
    disabled={false}
    disabledKeyboardNavigation={false}
    dropdownMode="scroll"
    maxDate={"2018-01-21T00:00:00.000Z"}
    minDate={"2018-01-17T00:00:00.000Z"}
    monthsShown={1}
    onBlur={[Function]}
    onChange={[Function]}
    onClickOutside={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMonthChange={[Function]}
    onSelect={[Function]}
    onYearChange={[Function]}
    placeholderText="18-01-2018"
    preventOpenOnFocus={false}
    shouldCloseOnSelect={true}
    showTimeSelect={false}
    timeIntervals={30}
    withPortal={false}
  />
</div>
`;

exports[`InputField Component matches snapshot 1`] = `
<label
  className="input"
  htmlFor="uniqueId"
>
  <span
    className="input--text"
  >
    From
  </span>
  <input
    id="uniqueId"
  />
</label>
`;

exports[`Range Component matches snapshot 1`] = `
<div
  className="search-bar--input-range"
>
  <span
    className="input input--text"
  >
    Refine Flight Search
  </span>
  <InputRange
    allowSameValues={false}
    classNames={
      Object {
        "activeTrack": "input-range__track input-range__track--active",
        "disabledInputRange": "input-range input-range--disabled",
        "inputRange": "input-range",
        "labelContainer": "input-range__label-container",
        "maxLabel": "input-range__label input-range__label--max",
        "minLabel": "input-range__label input-range__label--min",
        "slider": "input-range__slider",
        "sliderContainer": "input-range__slider-container",
        "track": "input-range__track input-range__track--background",
        "valueLabel": "input-range__label input-range__label--value",
      }
    }
    disabled={false}
    maxValue={200}
    minValue={0}
    onChange={[Function]}
    step={10}
    value={
      Object {
        "max": 200,
        "min": 0,
      }
    }
  />
</div>
`;

exports[`SearchBar Component matches snapshot 1`] = `
<div
  className="search-bar"
>
  <div
    className="search-bar--buttons"
  >
    <Button
      className="button--search-bar one-way"
      handleClick={[Function]}
      htmlFor="one-way"
      value="One way"
    />
    <Button
      className="button--search-bar return active"
      handleClick={[Function]}
      htmlFor="return"
      value="Return"
    />
  </div>
  <Form
    departureDate="18-01-2018"
    handleClicksOnSearch={[Function]}
    handleDepartureDateChange={[Function]}
    handleReturnDateChange={[Function]}
    isReturnFlight={true}
    onDestinationChange={[Function]}
    onFromChange={[Function]}
    onPassengerNumberChange={[Function]}
    returnDate="20-01-2018"
  />
  <Range
    handleRangeValueChange={[Function]}
    rangeValue={
      Object {
        "max": 200,
        "min": 0,
      }
    }
  />
</div>
`;
